Lib/subprocess.py

```diff
@@ -1235,8 +1235,10 @@ def communicate(self, input=None, timeout=None):
 
             finally:
                 self._communication_started = True
-
-            sts = self.wait(timeout=self._remaining_time(endtime))
+            try:
+                sts = self.wait(timeout=self._remaining_time(endtime))
+            except TimeoutExpired:
+                raise TimeoutExpired(self.args, timeout)
```

Comment:
I modified these to not create a new one, just reassign a new timeout value to the existing exception instance and reraise it.